### DISCUSSION TITLE
How to apply a price list on products ?

### DISCUSSION THREAD

**Discussion by ayoubqrt** [medusajs/medusa#2723] [Q&A]
Hey everyone ! I wanted these days to test medusa and specially the medusa admin interface. I struggled with the "Price lists". I have created a price list for example the black friday, then I lowered the price for a product from 12 dollars to 1 (there is only one variant for this product).

![image](https://user-images.githubusercontent.com/57446997/205463801-1b8b13ed-32ae-48fb-a768-162307322e25.png)

![image](https://user-images.githubusercontent.com/57446997/205464006-44e0df09-b0c9-490d-93c9-1a68b2eda362.png)

(The price list is published and not a draft)
![image](https://user-images.githubusercontent.com/57446997/205463794-229accde-8527-4cbe-ba2f-1a8353924557.png)

When I come back to the product and see the raw api response, the price hasn't changed and he's still at 12 dollars. 
![image](https://user-images.githubusercontent.com/57446997/205463814-b3261936-d789-4d1b-955d-f890ba412945.png)

And it's the same when I check the api response in my storefront, the price is higher because of the 20% taxes. 
![image](https://user-images.githubusercontent.com/57446997/205464685-a701b230-3a7a-47b1-bc63-3bd59fa43170.png)

Am I doing something wrong or it's a bug ? 

Thanks for you reading (and sorry for the big screenshots)

**Comment by pKorsholm** [comment]
Hi @ayoubqrt 

A couple of questions: 
- Do you see your price-list-price in the `prices` array in your storefront response? 
- are you querying for products using a cart_id/region_id or currency_code. Generally in order to apply the price selection strategy one of these parameters should be used when querying for products in order to get calculated prices.

**Comment by ayoubqrt** [comment]
Hi @pKorsholm, sorry for the late response.

I was on railway and to keep my free credits, I have deleted the project so today I have reployed the backend on railway, and magically in the store front I can see now the new prices overrided !

![image](https://user-images.githubusercontent.com/57446997/206897722-02018cfc-1a7a-4c91-972f-945b8544b2d8.png)

To answer your questions, I was using the medusa storefront nextjs starter and by default I think when the cart is created there is the region_id. But last time when I was looking in the api response, the `prices` array was empty for sure.

So I don't know what I have made wrong before and maybe do you have fix something ?

**Reply by ayoubqrt** [reply]
correction : 
"the `prices` array was empty for sure." => the `price-list` and `price-list-id` properties in the `prices` array was empty

**Reply by ayoubqrt** [reply]
Now I tried to change the default currency from usd to eur and it's still working so fine !
